src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -516,27 +518,25 @@ class Polyhedron
       int face_count;
       getFaces(faces, face_size, face_offset, face_count);
 
-      VectorType origin(m_vertices[0].data());
+      const PointType& origin = m_vertices[0];
 
       for(int i = 0; i < face_count; i++)
       {
         int n = face_size[i];
-        VectorType v0(m_vertices[faces[face_offset[i]]].data());
-        v0 -= origin;
+        VectorType v0 = m_vertices[faces[face_offset[i]]] - origin;
 
         for(int j = 1; j < n - 1; ++j)
         {
-          VectorType v1(m_vertices[faces[face_offset[i] + j]].data());
-          v1 -= origin;
-          VectorType v2(m_vertices[faces[face_offset[i] + ((j + 1) % n)]].data());
-          v2 -= origin;
-          double partialVol = v0.dot(VectorType::cross_product(v1, v2));
-          retVol += partialVol;
+          const int k = (j + 1) < n ? j + 1 : 0;
```

Comment:
Why is the check needed? Given the range of the for-loop, j <= n-2. So j+1 < n always. Thus, k = j + 1 always. Correct?

Reply:
Good call -- I updated the code.

I wrote some additional tests to verify the updated `Polyhedron::volume()` implementation. As part of this, I also added new functions -- `primal::Polygon::area()` and `primal::Polygon::signedArea()` (and associated tests).